src/traffic_layout/traffic_layout.cc

```diff
@@ -21,158 +21,67 @@
-  appVersionInfo.setup(PACKAGE_NAME, "traffic_layout", PACKAGE_VERSION, __DATE__, __TIME__, BUILD_MACHINE, BUILD_PERSON, "");
-  // Process command line arguments and dump into variables
-  if (!process_args_ex(&appVersionInfo, argument_descriptions, countof(argument_descriptions), argv) ||
-      file_arguments[0] != nullptr) {
-    help_usage();
-  }
-
-  runroot_handler(argv, 0 != cl.json);
-
-  if (cl.features) {
-    produce_features(0 != cl.json);
-  } else {
-    produce_layout(0 != cl.json);
-  }
-  return 0;
-}
-
-// handle everything with runroot using engine
-int
-traffic_runroot(int argc, const char **argv)
+main(int argc, const char **argv)
 {
-  // runroot engine for operations
-  RunrootEngine engine;
+  LayoutEngine engine;
 
   int i = 0;
   while (argv[i]) {
     engine._argv.push_back(argv[i]);
     ++i;
   }
-  // parse the command line & put into global variable
-  if (!engine.runroot_parse()) {
-    engine.runroot_help_message(true, true, true);
-    return 0;
-  }
-  // check sanity of the command about the runroot program
-  engine.sanity_check();
-
-  // create layout for runroot handling
-  runroot_handler(argv);
+  engine.parser.add_global_usage("traffic_layout CMD [OPTIONS]");
+
+  // global options
+  engine.parser.add_option("--features", "-f", "Show the compiled features");
+  engine.parser.add_option("--json", "-j", "Produce output in JSON format (when supported)");
+  engine.parser.add_option("--version", "-V", "Print version string");
+  engine.parser.add_option("--help", "-h", "Print usage information");
+  engine.parser.add_option("--run-root", "", "using TS_RUNROOT as sandbox", "", 1);
+
+  // info command
+  engine.parser.add_command("info", "Show the layout as default");
+  // init command
+  engine.parser.add_command("init", "Initialize(create) the runroot sandbox")
+    .add_option("--absolute", "", "Produce absolute path in the runroot.yaml")
+    .add_option("--force", "", "Create runroot even if the directory is not empty")
+    .add_option("--path", "", "Specify the path of the runroot", "", 1)
+    .add_option("--copy-style", "", "Specify the way of copying (FULL/HARD/SOFT)", "", 1)
+    .add_option("--layout", "", "Use specific layout (providing YAML file) to create runroot", "", 1);
+  // remove command
+  engine.parser.add_command("remove", "Remove the runroot sandbox")
+    .add_option("--force", "", "Remove runroot even if runroot.yaml is not found")
+    .add_option("--path", "", "Specify the path of the runroot", "", 1);
+  // verify command
+  engine.parser.add_command("verify", "Verify the runroot permissions")
+    .add_option("--fix", "", "Fix the permission issues of runroot")
+    .add_option("--path", "", "Specify the path of the runroot", "", 1);
+
+  engine.arguments = engine.parser.parse(argv);
+
+  runroot_handler(argv, engine.arguments.get("json"));
   Layout::create();
 
-  // check the command to execute
-  if (engine.run_flag) {
+  if (engine.arguments.get("init")) {
```

Comment:
Why is this here? I thought you could install these functions directly in the argument parser.